response_operations_ui/contexts/reporting_units.py

```diff
@@ -0,0 +1,199 @@
+        row = {
+            "status_class": _select_status_class(ce["responseStatus"]),
+            "hyperlink": url_for(
+                "case_bp.get_response_statuses",
+                ru_ref=ru["sampleUnitRef"],
+                survey=survey["shortName"],
+                period=ce["exerciseRef"],
+            ),
+            "hyperlink_text": "Change" if permissions[0] else "View",
+            "period": ce["exerciseRef"],
+            "reporting_unit_name": ce["companyName"],
+            "trading_as": ce["tradingAs"],
+            "region": ce["companyRegion"],
+            "response_status": ce["responseStatus"],
+            "status": _build_ce_status(ru, survey, ce, permissions),
+        }
+        table.append(row)
+
+    return table
+
+
+def _select_status_class(status):
+    if status == "Not started":
+        return "ons-status--info"
+    elif status == "In progress":
+        return "ons-status--pending"
+    elif status == "Completed" or status == "Completed by phone":
+        return "ons-status--success"
+    elif status == "No longer required":
+        return "ons-status--dead"
+    else:
+        return "ons-status--error"
+
+
+def _build_ce_status(ru, survey, ce, permissions):
+    hyperlink = url_for(
+        "case_bp.get_response_statuses",
+        ru_ref=ru["sampleUnitRef"],
+        survey=survey["shortName"],
+        period=ce["exerciseRef"],
+    )
+    hyperlink_text = "Change" if permissions[0] else "View"
+    status_class = _select_status_class(ce["responseStatus"])
+    response_status = ce["responseStatus"]
+    return (
+        '<span class="ons-status '
+        + status_class
+        + '">'
+        + response_status
+        + "</span>"
+        + "&nbsp;   "
+        + '<a href="'
+        + hyperlink
+        + '">'
+        + hyperlink_text
+        + "</a>"
+    )
+
+
+def _build_respondents_section(
+    respondents: list, case, collection_exercises: list, ru: dict, survey: dict, unused_iac: str, permissions: list
+):
+    table = []
+    for respondent in respondents:
+        row = {}
+        if permissions[0]:
+            if unused_iac:
+                row["enrolment_code"] = unused_iac
+            else:
+                row["enrolment_code_hyperlink"] = url_for(
+                    "reporting_unit_bp.generate_new_enrolment_code",
+                    case_id=case["id"],
+                    collection_exercise_id=collection_exercises[0]["id"],
+                    ru_name=ru["name"],
+                    ru_ref=ru["sampleUnitRef"],
+                    trading_as=collection_exercises[0]["tradingAs"],
+                    survey_ref=survey["surveyRef"],
+                    survey_name=survey["shortName"],
+                )
+                row["enrolment_code_hyperlink_text"] = "Generate new enrollment code"
+        else:
+            row["enrolment_code"] = ""
+        row["contact_details"] = {
+            "Name": f"{respondent['firstName']} {respondent['lastName']}",
```

Comment:
Are you using the key in the frontend as I usually wouldn't expect capitalised keys

Reply:
I was going to and then didn't this has been updated